operations/mimir-mixin/alerts/alerts.libsonnet

```diff
@@ -221,6 +221,42 @@
             message: '{{ $labels.job }}/%(alert_instance_variable)s has a number of mmap-ed areas close to the limit.' % $._config,
           },
         },
+      ] + [
+        {
+          alert: $.alertName('RingMembersMismatch'),
+          expr: |||
+            (
+              avg by(%(alert_aggregation_labels)s) (sum by(%(alert_aggregation_labels)s, %(per_instance_label)s) (cortex_ring_members{name="%(component)s",job=~"(.*/)?%(job)s"}))
+              != sum by(%(alert_aggregation_labels)s) (up{job=~"(.*/)?%(job)s"})
+            )
+            and
+            (
+              count by(%(alert_aggregation_labels)s) (cortex_build_info) > 0
+            )
+          ||| % {
+            alert_aggregation_labels: $._config.alert_aggregation_labels,
+            per_instance_label: $._config.per_instance_label,
+            component: component_job[0],
+            job: component_job[1],
+          },
+          'for': '15m',
+          labels: {
+            component: component_job[0],
+            severity: 'warning',
+          },
+          annotations: {
+            message: |||
+              Number of members in Mimir %(component)s hash ring does not match the expected number in %(alert_aggregation_variables)s.
+            ||| % { component: component_job[0], alert_aggregation_variables: $._config.alert_aggregation_variables },
+          },
+        }
+        for component_job in [
```

Comment:
I did some backtesting of this alert (which I also suggested you to do to validate it) and it would fire too often for some stateless components (e.g. distributors). To avoid introducing a spammy alert we could focus on the only critical component here, which is the ingester, and have the alert only for the ingester at this stage.

A part from this, everything is in the alert definition looks good! 👏

Reply:
I did some spot checks and didn't think it was firing too much. Perhaps I'm looking at it wrong though. Would it be worthwhile to have the other components checked but maybe for a longer time?

Reply:
Not sure. In some cells it would fire even if you increase `for` duration. In my opinion what's really critical here is the ingesters ring. Other ring are OK-ish if don't match (especially for stateless services which are prone to OOM).